plugin-gradle/src/test/java/com/diffplug/gradle/spotless/JsonExtensionTest.java

```diff
@@ -0,0 +1,220 @@
+						"    target 'examples/**/*.json'" +
+						"    simple()" +
+						"}",
+				"}");
+		setFile("src/main/resources/example.json").toResource("json/nestedObjectBefore.json");
+		setFile("examples/main/resources/example.json").toResource("json/nestedObjectBefore.json");
+		gradleRunner().withArguments("spotlessApply").build();
+		assertFile("src/main/resources/example.json").sameAsResource("json/nestedObjectBefore.json");
+		assertFile("examples/main/resources/example.json").sameAsResource("json/nestedObjectAfter.json");
+	}
+
+	@Test
+	public void formattingWithCustomNumberOfSpaces() throws IOException {
+		setFile("build.gradle").toLines(
+				"buildscript { repositories { mavenCentral() } }",
+				"plugins {",
+				"    id 'java'",
+				"    id 'com.diffplug.spotless'",
+				"}",
+				"spotless {",
+				"    json {" +
+						"    target '**/*.json'" +
+						"    simple {" +
+						"        indentWithSpaces(6)" +
+						"    }",
+				"}",
+				"}");
+		setFile("src/main/resources/example.json").toResource("json/singletonArrayBefore.json");
+		gradleRunner().withArguments("spotlessApply").build();
+		assertFile("src/main/resources/example.json").sameAsResource("json/singletonArrayAfter6Spaces.json");
+	}
+
+	@Test
+	public void formattingWithCustomNumberOfSpaces___() throws IOException {
+		setFile("build.gradle").toLines(
+				"buildscript { repositories { mavenCentral() } }",
+				"plugins {",
+				"    id 'java'",
+				"    id 'com.diffplug.spotless'",
+				"}",
+				"spotless {",
+				"    json {" +
+						"    target '**/*.json'" +
+						"    simple().indentWithSpaces(6)" +
+						"}",
+				"}");
+		setFile("src/main/resources/example.json").toResource("json/singletonArrayBefore.json");
+		gradleRunner().withArguments("spotlessApply").build();
+		assertFile("src/main/resources/example.json").sameAsResource("json/singletonArrayAfter6Spaces.json");
+	}
+
+	@Test
+	public void formattingWithCustomNumberOfSpaces___format() throws IOException {
+		setFile("build.gradle").toLines(
+				"buildscript { repositories { mavenCentral() } }",
+				"plugins {",
+				"    id 'java'",
+				"    id 'com.diffplug.spotless'",
+				"}",
+				"spotless {",
+				"    format 'json', {" +
+						"    target '**/*.json'" +
+						"    simple().indentWithSpaces(6)" +
+						"}",
+				"}");
+		setFile("src/main/resources/example.json").toResource("json/singletonArrayBefore.json");
+		gradleRunner().withArguments("spotlessApply").build();
+		assertFile("src/main/resources/example.json").sameAsResource("json/singletonArrayAfter6Spaces.json");
+	}
+
+	@Test
+	public void formattingWithCustomNumberOfSpaces___formatWithClass() throws IOException {
+		setFile("build.gradle").toLines(
+				"buildscript { repositories { mavenCentral() } }",
+				"plugins {",
+				"    id 'java'",
+				"    id 'com.diffplug.spotless'",
+				"}",
+				"spotless {",
+				"    format 'json', com.diffplug.gradle.spotless.JsonExtension, {" +
+						"    target '**/*.json'" +
```

Comment:
Lots of tests have this problem: replace ` + ` with `,`. You are building this string:

```
spotless {
  format 'json', com.diffplug.gradle.spotless.JsonExtension, { target '**/*.json' simple().intendWithSpaces } 
```

When you mean to build this one:

```
spotless {
  format 'json', com.diffplug.gradle.spotless.JsonExtension, {
    target '**/*.json'
    simple().intendWithSpaces
  }
}
```

Reply:
Oh, doh! :see_no_evil: Yep that fixes tests that were previously failing, great spot